pennylane/wires.py

```diff
@@ -384,15 +399,21 @@ def all_wires(list_of_wires, sort=False):
         >>> Wires.all_wires(list_of_wires)
         <Wires = [4, 0, 1, 3, 5]>
         """
-
-        combined = []
         for wires in list_of_wires:
             if not isinstance(wires, Wires):
                 raise WireError(
                     "Expected a Wires object; got {} of type {}".format(wires, type(wires))
                 )
 
-            combined.extend(wire for wire in wires.labels if wire not in combined)
+        label_sets = map(lambda x: x.toset(), list_of_wires)
+        # This makes a set of all labels in O(n) time.
+        all_labels = functools.reduce(lambda a, b: a | b, label_sets, set())
+        combined = []
+        for wires in list_of_wires:
+            extension = [wire for wire in wires.labels if wire in all_labels]
```

Comment:
Is the `if` necessary here? Won't all `wire`s appearing here be in `all_labels` already?